### PR TITLE
glTF2 use alpha, alpha to coverage and depth prepass

### PR DESCRIPTION
**Pull request by fire**
0bdbce672

For https://github.com/KhronosGroup/glTF-Sample-Models/tree/master/2.0/AlphaBlendModeTest cases.

image

image

Fixes the second case of blending from gltf2 test case.

image

image

Fixes hair for the the blend case.

Not fixed in the PR: Alpha being order independent transparency.

### CODE REVIEW

File: modules/gltf/gltf_document.cpp [hunk 1]
```diff
@@ -3519,17 +3524,21 @@ Error GLTFDocument::_parse_materials(Ref<GLTFState> state) {
...
 		if (d.has("alphaMode")) {
 			const String &am = d["alphaMode"];
+			material->set_depth_draw_mode(BaseMaterial3D::DEPTH_DRAW_ALWAYS);
 			if (am == "BLEND") {
-				material->set_transparency(BaseMaterial3D::TRANSPARENCY_ALPHA_DEPTH_PRE_PASS);
+				material->set_transparency(BaseMaterial3D::TRANSPARENCY_ALPHA);
 			} else if (am == "MASK") {
+				float alpha_cutoff = 0.5f;
 				material->set_transparency(BaseMaterial3D::TRANSPARENCY_ALPHA_SCISSOR);
+				material->set_alpha_antialiasing(BaseMaterial3D::ALPHA_ANTIALIASING_ALPHA_TO_COVERAGE_AND_TO_ONE);
```
Comment (clayjohn) [inline modules/gltf/gltf_document.cpp:3535]: I'm not sure if this is the best approach. Won't this only work when MSAA is enabled? E.g. when alpha to coverage is available?

Comment (fire) [reply modules/gltf/gltf_document.cpp:3535]: According to the source designs, it works when aa is off.

https://github.com/godotengine/godot/pull/40364

https://bgolus.medium.com/anti-aliased-alpha-test-the-esoteric-alpha-to-coverage-8b177335ae4f

\<Ben Golus> In fact if MSAA is disabled Alpha to Coverage falls back to alpha test. This means it isn’t only useful for VR or other games where you’re guaranteed to have MSAA enabled. Basically there’s rarely a reason to use straight alpha testing ever!

**Comment by fire** [comment]
Need to evaluate again after https://github.com/godotengine/godot/pull/55084 and https://github.com/godotengine/godot/pull/55085

**Comment by fire** [comment]
Closing for now due to need to reevaluate.

**Pull request closed without merging**